lib/api/core/models/repositories/profileRepository.js

```diff
@@ -205,28 +188,7 @@ module.exports = function (kuzzle) {
    * @returns {Object}
    */
   ProfileRepository.prototype.serializeToDatabase = function (profile) {
-    var result = {};
-
-    Object.keys(profile).forEach(function (key) {
-      if (key !== 'roles') {
-        result[key] = profile[key];
-      }
-    });
-
-    result.roles = [];
-    profile.roles.forEach(role => {
-      var serializedRole = {_id: role};
-      if (_.isObject(role)) {
-        serializedRole = {_id: role._id};
-      }
-      if (!_.isEmpty(role.restrictedTo)) {
-        serializedRole.restrictedTo = role.restrictedTo;
-      }
-      if (role.allowInternalIndex) {
-        serializedRole.allowInternalIndex = role.allowInternalIndex;
-      }
-      result.roles.push(serializedRole);
-    });
+    var result = _.assignIn({}, profile);
```

Comment:
nitpicking: following our discussion: could you add a small comment explaining that this line is used to avoid the mutation of the `profile` object?